src/bootstrap/test.rs

```diff
@@ -1003,7 +1003,7 @@ note: if you're sure you want to do this, please open an issue as to why. In the
 
         // Avoid depending on rustdoc when we don't need it.
         if mode == "rustdoc"
-            || (mode == "run-make" && suite.ends_with("fulldeps"))
+            || mode == "run-make"
```

Comment:
I would prefer to avoid this and instead move the test to fulldeps.

Reply:
That seems a shame when this doesn't actually need a stage 2 build :/ that means you have to compiler rustc twice to run the test at all. Rustdoc only takes about a minute to build even with a clean cache.

Reply:
I don't want to make this change in this PR. We can consider it separately.

Reply:
Ok, I reverted the change. I do think it would be useful though - https://github.com/rust-lang/rust/pull/81223 could also use it, and that test is much more likely to fail, so building the compiler twice is painful.